src/VisualStudio.Roslyn.SDK/ComponentDebugger/DebugProfileProvider.cs

```diff
@@ -0,0 +1,95 @@
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+using Microsoft.VisualStudio.ProjectSystem.VS.Debug;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
+
+namespace Roslyn.ComponentDebugger
+{
+    [Export(typeof(IDebugProfileLaunchTargetsProvider))]
+    [AppliesTo(Constants.RoslynComponentCapability)]
+    public class DebugProfileProvider : IDebugProfileLaunchTargetsProvider
+    {
+        private readonly ConfiguredProject _configuredProject;
+
+        private readonly string _compilerRoot;
+
+        [ImportingConstructor]
+        [Obsolete("This exported object must be obtained through the MEF export provider.", error: true)]
+        public DebugProfileProvider(ConfiguredProject configuredProject, SVsServiceProvider? serviceProvider)
+        {
+            _configuredProject = configuredProject;
+            _compilerRoot = GetCompilerRoot(serviceProvider);
+        }
+
+        public Task OnAfterLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile) => Task.CompletedTask;
+
+        public Task OnBeforeLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile) => Task.CompletedTask;
+
+        public bool SupportsProfile(ILaunchProfile? profile) => Constants.CommandName.Equals(profile?.CommandName, StringComparison.OrdinalIgnoreCase);
+
+        public async Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsAsync(DebugLaunchOptions launchOptions, ILaunchProfile? profile)
+        {
+            // set up the managed (net fx) debugger to start a process
+            var settings = new DebugLaunchSettings(launchOptions)
+            {
+                LaunchDebugEngineGuid = Microsoft.VisualStudio.ProjectSystem.Debug.DebuggerEngines.ManagedOnlyEngine,
+                LaunchOperation = DebugLaunchOperation.CreateProcess
+            };
+
+            // try and get the target project
+            if (TryGetTargetProject(_configuredProject, profile, out var targetProjectUnconfigured))
+            {
+                settings.CurrentDirectory = Path.GetDirectoryName(targetProjectUnconfigured!.FullPath);
+                var compiler = _configuredProject.Capabilities.Contains(ProjectCapabilities.VB) ? "vbc.exe" : "csc.exe";
+                settings.Executable = Path.Combine(_compilerRoot, compiler);
+
+                // try and get the configured version of the target project
+                var targetProject = await targetProjectUnconfigured.GetSuggestedConfiguredProjectAsync().ConfigureAwait(false);
+                if (targetProject is object)
+                {
+                    // get its compilation args
+                    var args = await targetProject.GetCompilationArgumentsAsync().ConfigureAwait(false);
+
+                    // append the command line args to the debugger launch
+                    settings.Arguments = string.Join(" ", args);
```

Comment:
Curious: if there are say spaces in soucrce file paths will the `args` local already contain `"` around them?

Reply:
Yes, each argument is correctly quoted (and I double checked with a test run)

---
In reply to: [573355409](https://github.com/dotnet/roslyn-sdk/pull/726#discussion_r573355409) [](ancestors = 573355409)